owl/SOMA-IO.owl

```diff
@@ -25,40 +124,321 @@
+            </owl:Restriction>
+        </rdfs:subClassOf>
+        <rdfs:comment xml:lang="en">An Archive Format is the file format of an archive file.
+
+Source: https://en.wikipedia.org/wiki/Archive_file#Archive_formats</rdfs:comment>
+        <rdfs:label xml:lang="en">Archive Format</rdfs:label>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#ArtificialAgent -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#ArtificialAgent">
+        <rdfs:subClassOf rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#PhysicalAgent"/>
+        <rdfs:comment xml:lang="en">Any constructed agent with physical embodiment that has intentions on its own, e.g., a robot.</rdfs:comment>
+        <rdfs:label xml:lang="en">Artificial agent</rdfs:label>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#Classifier -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#Classifier">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#StatisticalReasoner"/>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#Client-Server_Specification -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#Client-Server_Specification">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#InterfaceSpecification"/>
+        <rdfs:subClassOf>
+            <owl:Class>
+                <owl:intersectionOf rdf:parseType="Collection">
+                    <owl:Restriction>
+                        <owl:onProperty rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#definesRole"/>
+                        <owl:someValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#ClientRole"/>
+                    </owl:Restriction>
+                    <owl:Restriction>
+                        <owl:onProperty rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#definesRole"/>
+                        <owl:someValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#ServerRole"/>
+                    </owl:Restriction>
+                </owl:intersectionOf>
+            </owl:Class>
+        </rdfs:subClassOf>
+        <rdfs:label xml:lang="en">Client-Server Specification</rdfs:label>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#ClientRole -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#ClientRole">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#InterfaceComponentRole"/>
+        <rdfs:subClassOf>
+            <owl:Restriction>
+                <owl:onProperty rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#isDefinedIn"/>
+                <owl:someValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#Client-Server_Specification"/>
+            </owl:Restriction>
+        </rdfs:subClassOf>
+        <rdfs:label xml:lang="en">Client role</rdfs:label>
+    </owl:Class>
```

Comment:
Some comment here would be nice.

More importantly, a question: would it make sense to commit to a more strict definition of this role in terms of what it can apply to? E.g. a piece of stone might not qualify as an entity filling this role, and presumably neither would the concept of money.

Reply:
I added the restriction that Software Roles only classsify Software.